pandas/tests/dtypes/test_cast.py

```diff
@@ -407,3 +408,20 @@ def test_period_dtype(self):
                        np.dtype('datetime64[ns]'), np.object, np.int64]:
             assert find_common_type([dtype, dtype2]) == np.object
             assert find_common_type([dtype2, dtype]) == np.object
+
+    @pytest.mark.parametrize('dtype', [int, float, str, object, None])
+    @pytest.mark.parametrize('datum1', [1, 2., "3", (4, 5), [6, 7], None])
+    @pytest.mark.parametrize('datum2', [8, 9., "10", (11, 12), [13, 14], None])
+    def test_cast_1d_array(self, dtype, datum1, datum2):
+        data = [datum1, datum2]
+        try:
+            problem = (dtype is not object and
+                       np.array(data, dtype=dtype).ndim != 1) and ValueError
+        except (ValueError, TypeError) as exc:
```

Comment:
can you make 2 tests, 1 of which all succeed, 1 of which tests the ones that error

Reply:
I can, but then I wouldn't parametrize (that would be a nightmare, since the cartesian product of parameters cannot be cleanly split). Let me try to add some comments to make it more clear instead.